spring-cloud-gateway-server/src/test/java/org/springframework/cloud/gateway/filter/headers/ForwardedHeadersFilterTests.java

```diff
@@ -195,4 +195,42 @@ public void forwardedParsedCorrectly() {
 		}
 	}
 
+	@Test
+	public void forwardedByForIpv4AddressIsAdded() throws UnknownHostException {
+		Forwarded forwarded = new Forwarded();
+		InetAddress ipv4Address = InetAddress.getByName("216.103.69.111");
+		ForwardedHeadersFilter forwardedHeadersFilter = new ForwardedHeadersFilter();
+		forwardedHeadersFilter.setByEnabled(true);
+
+		forwardedHeadersFilter.addForwardedBy(forwarded, ipv4Address);
+
+		Assertions.assertThat(forwarded.getValues()).containsEntry("by", "216.103.69.111");
+
+	}
+
+	@Test
+	public void forwardedByForIpv6AddressIsAdded() throws UnknownHostException {
+		Forwarded forwarded = new Forwarded();
+		InetAddress ipv6Address = InetAddress.getByName("abc4:babf:955f:1724:11bc:0153:275c:d36e");
+		ForwardedHeadersFilter forwardedHeadersFilter = new ForwardedHeadersFilter();
+		forwardedHeadersFilter.setByEnabled(true);
+
+		forwardedHeadersFilter.addForwardedBy(forwarded, ipv6Address);
+
+		Assertions.assertThat(forwarded.getValues()).containsEntry("by",
+				"\"[abc4:babf:955f:1724:11bc:153:275c:d36e]\"");
+
+	}
+
+	@Test
+	public void forwardedByIsNotAddedIfFeatureIsDisabled() throws UnknownHostException {
+		Forwarded forwarded = new Forwarded();
+		InetAddress ipv4Address = InetAddress.getByName("216.103.69.111");
+		ForwardedHeadersFilter forwardedHeadersFilter = new ForwardedHeadersFilter();
+		forwardedHeadersFilter.setByEnabled(false);
+
+		forwardedHeadersFilter.addForwardedBy(forwarded, ipv4Address);
+
+		Assertions.assertThat(forwarded.getValues()).containsEntry("by", "216.103.69.111");
```

Comment:
I don't understand this test. You manually call "addForwardedBy" which is not checking if the by-feature is enabled and due to this the "by" value is present.

Reply:
yup, that seems to be a mistake